### PR TITLE
PPTP-426: view registration implementation

### PR DESCRIPTION
**Pull request by tombart**

### CODE REVIEW

**Review by tmc-ee — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtax/returns/config/AppConfig.scala [hunk 1]
```diff
@@ -60,6 +60,9 @@ class AppConfig @Inject() (config: Configuration, val servicesConfig: ServicesCo
 
   def pptReturnUrl(id: String): String = s"$pptReturnsUrl/$id"
 
+  def pptGetRegistrationDetailsUrl(pptReference: String): String =
+    s"$pptServiceHost/registrations/$pptReference"
+
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/returns/config/AppConfig.scala:65]: :thinking: I think on the returns service we should always reference it as `subscriptions` and not `registrations`
The call to EIS/ETMP is a subscriptions display which is the domain language we use and I believe that domain language should continue on our code, as even I got a bit confused here :)

File: app/uk/gov/hmrc/plasticpackagingtax/returns/connectors/RegistrationsConnector.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import uk.gov.hmrc.http.HttpReads.Implicits.readFromJson
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import uk.gov.hmrc.plasticpackagingtax.returns.config.AppConfig
+import uk.gov.hmrc.plasticpackagingtax.returns.models.registration._
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RegistrationsConnector @Inject() (
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/returns/connectors/RegistrationsConnector.scala:29]: It should be called `SubscriptionsConnector`

File: app/uk/gov/hmrc/plasticpackagingtax/returns/connectors/RegistrationsConnector.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    httpClient.GET[RegistrationDetails](appConfig.pptGetRegistrationDetailsUrl(pptReferenceNumber))
+      .andThen { case _ => timer.stop() }
+      .recover {
+        case exception: Exception =>
+          throw DownstreamServiceError(
+            s"Failed to retrieve registration details for PPTReference: [$pptReferenceNumber], error: [${exception.getMessage}]",
+            exception
+          )
+      }
+  }
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/returns/connectors/RegistrationsConnector.scala:46]: I like this! :)

File: app/uk/gov/hmrc/plasticpackagingtax/returns/controllers/home/ViewRegistrationDetailsController.scala [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import uk.gov.hmrc.plasticpackagingtax.returns.controllers.home.{routes => homeRoutes}
+import uk.gov.hmrc.plasticpackagingtax.returns.models.request.JourneyAction
+import uk.gov.hmrc.plasticpackagingtax.returns.views.html.home.view_registration_details_page
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ViewRegistrationDetailsController @Inject() (
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/returns/controllers/home/ViewRegistrationDetailsController.scala:33]: Please rename it to `Subscriptions`

**Review by PauloPortugal — APPROVED** [review]
looks good. 
Please just update the domain language from Registrations to Subscriptions including API resources.

File: app/uk/gov/hmrc/plasticpackagingtax/returns/config/AppConfig.scala (hunk 1 above)
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/returns/config/AppConfig.scala]: done

File: app/uk/gov/hmrc/plasticpackagingtax/returns/controllers/home/ViewRegistrationDetailsController.scala (hunk 4 above)
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/returns/controllers/home/ViewRegistrationDetailsController.scala]: done

**Pull request merged**